### PR TITLE
KATA-681: Understanding section: moving doc from Google Docs to GitHub

### PR DESCRIPTION
**Pull request by sagidlow**
JIRA Link: https://issues.redhat.com/browse/KATA-681
Applies to 4.8+
Preview Link: https://deploy-preview-32307--osdocs.netlify.app/openshift-enterprise/latest/sandboxed_containers/understanding-sandboxed-containers.html 

**Preliminary doc is in Google Doc and we are in the process of moving it from Google to GitHub. This is a part of the 2nd draft.**

### CODE REVIEW

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:9]: Thought: this would be a great place for [description lists](https://asciidoctor.org/docs/asciidoc-writers-guide/#description-lists).

**Comment by maxwelldb** [comment]
> Preliminary doc is in Google Doc and we are in the process of moving it from Google to GitHub. This is a part of the 2nd draft.

Is this PR as complete as it's going to be for review purposes? @sagidlow

File: modules/sandboxed-containers-common-terms.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-common-terms.adoc:9]: This is a great idea! I didn't think about description list. I think this is a good idea as well.

File: sandboxed_containers/understanding-sandboxed-containers.adoc
Comment (maxwelldb) [inline sandboxed_containers/understanding-sandboxed-containers.adoc:19]: Translation

```suggestion
{sandboxed-containers-first} also provides users the ability to choose from the type of workload that they want to run to cover a wide variety of use cases.
```

File: sandboxed_containers/understanding-sandboxed-containers.adoc
Comment (maxwelldb) [inline sandboxed_containers/understanding-sandboxed-containers.adoc:21]: ISG:

> ⚠️ enable _verb_
> Use to describe the action of making an interface control or product function available to users. Avoid stating
that inanimate objects grant abilities to people, as in "the product enables you to...". Whenever possible, use a
direct, user-focused alternative such as "you can use the product to..." or "with this product, users can...".

**Review by maxwelldb — CHANGES_REQUESTED** [review]
Made some suggestions and comments. 

Anything that refers to ISG or our contribution guidelines is a must fix, but otherwise, feel free to consider comments as just one person's opinion. 

LMK if you have any questions or concerns. Happy to re-check following new commits, if you like.

File: modules/sandboxed-containers-common-terms.adoc
Comment (kalexand-rh) [inline modules/sandboxed-containers-common-terms.adoc:10]: >! execute _verb_
Use only if a simpler term, such as "run", "issue", "start", or "enter" is inappropriate.

File: modules/sandboxed-containers-common-terms.adoc
Comment (kalexand-rh) [reply modules/sandboxed-containers-common-terms.adoc:12]: >! using _noun_
To avoid ambiguity, replace this gerund with either "by using" or "that uses".

File: modules/sandboxed-containers-common-terms.adoc
Comment (kalexand-rh) [reply modules/sandboxed-containers-common-terms.adoc:15]: We [almost never use ](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#product-name-and-version)plain "OpenShift." This should probably be `{product-title}`.

File: modules/sandboxed-containers-common-terms.adoc
Comment (kalexand-rh) [reply modules/sandboxed-containers-common-terms.adoc:19]: +1 - you can (and should) use the attributes everywhere but in the `_topic_map.yml` file. I see a couple of OpenShift sandboxed containers and RHCOS references in headings that should be swapped to attributes.

File: modules/sandboxed-containers-common-terms.adoc
Comment (kalexand-rh) [inline modules/sandboxed-containers-common-terms.adoc:28]: You could tighten this a little.
```suggestion
`KataConfig` represents a configuration of sandboxed containers. Information about the state of the cluster, for example the nodes on which the software was deployed, is in the `KataConfig` objects.
```

File: modules/sandboxed-containers-common-terms.adoc
Comment (kalexand-rh) [reply modules/sandboxed-containers-common-terms.adoc:31]:
```suggestion
{op-system-first} extensions are a mechanism to install optional {product-title} software. The {sandboxed-containers} uses this mechanism to deploy sandboxed containers on a cluster.
```

There are attributes for both first and subsequent use of RHCOS - {op-system-first} and {op-system}. Please use them throughout.

File: modules/common-attributes.adoc
Comment (kalexand-rh) [reply modules/common-attributes.adoc:30]: +1 to this - the attribute is plural, and its rendered text is singular, so the subject/verb agreement mismatch is tripping me up.

File: modules/sandboxed-containers-rhcos-extensions.adoc
Comment (kalexand-rh) [reply modules/sandboxed-containers-rhcos-extensions.adoc:11]: You need move this out into the assembly as an additional resource or remove the link. We generally can't have direct links in modules (for now) because we can't have a link from the customer portal to docs.openshift, and we need to keep the user in the same product version.

**Review by kalexand-rh — COMMENTED** [review]
I largely agree with Max and have a couple more comments. This change is looking good, though!

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [reply modules/sandboxed-containers-common-terms.adoc:15]: +1. I zeroed in on "Kubernetes" too quickly.

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [reply modules/sandboxed-containers-common-terms.adoc:31]: Yeah, I thought that these existed, but I couldn't find them in the common attributes file, so I assumed they'd been nixed. I blame my ophthalmologist appointment earlier in the day. :)

File: modules/sandboxed-containers-common-terms.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-common-terms.adoc:19]: @maxwelldb + @kalexand-rh I was leaving it off from headings because of PV2 compliance. PV2 checklist: Do not use {variables} in title/chapter names or IDs

Comment (vikram-redhat) [reply modules/sandboxed-containers-common-terms.adoc:19]: Yes, ignore the PV2 checklist for now. FWIW, that guidance is only temporary.

File: modules/sandboxed-containers-building-blocks.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-building-blocks.adoc:11]: Is "uses:"

- "is based on"
- "consumes"
- something else

Comment (maxwelldb) [inline modules/sandboxed-containers-building-blocks.adoc:11]: The `KataConfig` resource? Object? Custom resource? 

https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#api-object-object-references

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:9]:
```suggestion
Sandbox:: A sandbox is an isolated environment where programs can run. In a sandbox, you can run untested or untrusted programs without risking harm to the host machine or the operating system.
```

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:11]: Different from what? 

Consider breaking into two sentences.

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:13]: Could this definition compare and contrast pods (general) vs. pods (this document)? 

Also, do we need to call out that it's both k8s + OCP, or can k8s be dropped? Not sure, myself. :)

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:17]: Could this be a brief concept module instead of a term?

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:21]:
```suggestion
Kata Containers:: Kata Containers is a core upstream project that is used to build {sandboxed-containers-first}. {sandboxed-containers-first} integrate Kata Containers with {product-title}.
```

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:23]:
```suggestion
KataConfig:: `KataConfig` objects represents configurations of sandboxed containers. They store information about the state of the cluster, such as the nodes on which the software is deployed.
```

File: modules/sandboxed-containers-common-terms.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-common-terms.adoc:27]: Guidelines tweak: 

```suggestion
Runtime class:: A `RuntimeClass` object describes which runtime can be used to run a given workload. A runtime class that is named `Kata` is installed and deployed by the {sandboxed-containers-operator}. The runtime class contains information about the runtime that describes resources that the runtime needs to operate, such as the pod overhead.
```

File: modules/sandboxed-containers-rhcos-extensions.adoc
Comment (maxwelldb) [inline modules/sandboxed-containers-rhcos-extensions.adoc:9]: "uses" == "implements" || "is based on"?

Comment (maxwelldb) [inline modules/sandboxed-containers-rhcos-extensions.adoc:9]: Typo fix

```suggestion
The {sandboxed-containers-operator} uses the {op-system-first} extensions concept. The Kata binaries, RPMs, and QEMU are provided as extensions. You can enable them by using the `MachineConfig` resources that the Machine Config Operator provides.
```

File: sandboxed_containers/understanding-sandboxed-containers.adoc
Comment (maxwelldb) [inline sandboxed_containers/understanding-sandboxed-containers.adoc:21]: ⚠️ ISG: "Enables"
> Use to describe the action of making an interface control or product function available to users. Avoid stating
that inanimate objects grant abilities to people, as in "the product enables you to...". Whenever possible, use a
direct, user-focused alternative such as "you can use the product to..." or "with this product, users can...".

**Review by maxwelldb — COMMENTED** [review]
Looks copacetic. I left a lot of comments because I like to type my thoughts as I read. 🙈 

Aside from typo fixes, the comment with the guidelines link, and `RuntimeClass`, I believe they're all just optional opinion concerns from me.

File: sandboxed_containers/understanding-sandboxed-containers.adoc
Comment (maxwelldb) [reply sandboxed_containers/understanding-sandboxed-containers.adoc:19]: Makes for weird reading. I'm sorry. 😆

File: modules/sandboxed-containers-building-blocks.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-building-blocks.adoc:11]: @maxwelldb: The PM used "uses" here because that is the operator image. Maybe better to rephrase?

File: modules/sandboxed-containers-common-terms.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-common-terms.adoc:17]: @maxwelldb The Operator itself as a building blocks section. Do you think we should move this content there instead? 
@zanetworker thoughts?

**Review by maxwelldb — APPROVED** [review]

File: modules/sandboxed-containers-building-blocks.adoc
Comment (maxwelldb) [reply modules/sandboxed-containers-building-blocks.adoc]: Possibly. I was thinking of translation, here, but I'm not sure how much it matters, TBH.

**Comment by sagidlow** [comment]
Pending SME approval: @zanetworker 
Pending QE approval: @cpmeadors

File: modules/sandboxed-containers-common-terms.adoc
Comment (zanetworker) [reply modules/sandboxed-containers-common-terms.adoc:13]: Since OCP is not introducing a `pod` resource on its own and is in fact a rebase of Kubernetes (using it underneath), we initially thought it's okay to use kuberetes. It might be easier to reference an existing definition in OpenShift documentation to prevent though contention to map to a project vs. a product.

File: modules/sandboxed-containers-common-terms.adoc
Comment (zanetworker) [reply modules/sandboxed-containers-common-terms.adoc:17]: @maxwelldb the idea of introducing terms here is to make the user quickly familiar with concepts necessary to understand the product flow by consuming content from the same place.

File: modules/sandboxed-containers-common-terms.adoc
Comment (c3d) [reply modules/sandboxed-containers-common-terms.adoc:13]: I understand @zanetworker's rationale, but I think that it does not hurt to write "A pod is a Kubernetes, OpenShift and {product-title} construct...". It does not seem overly verbose.

Or alternatively, "A pod is a {product-title} construct inherited from Kubernetes and OpenShift that represent resources where containers can be deployed." This wording may help clarify the points made in the next sentence that we are explaining how they work in the case of sandboxed containers.

File: modules/sandboxed-containers-common-terms.adoc
Comment (c3d) [reply modules/sandboxed-containers-common-terms.adoc:17]: @maxwelldb I think the difference with the example you gave us is the much smaller amount of configuration and tuning we expect in the case of sandboxed containers. At the moment at least, I think if we created a concept module, it would be mostly empty.

@zanetworker Do you think this could change in the future?

File: modules/sandboxed-containers-uninstalling-kata-runtime.adoc
Comment (zanetworker) [inline modules/sandboxed-containers-uninstalling-kata-runtime.adoc:20]: @jensfr @bpradipt can you ack this behavior?

File: modules/sandboxed-containers-building-blocks.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-building-blocks.adoc]: I'm just going to resolve this for now. I tried to rework the sentence but nothing I came up with made any sense >.<

**Review by zanetworker — COMMENTED** [review]
Pending reviews from operator code owners.

File: modules/sandboxed-containers-uninstalling-kata-runtime.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-uninstalling-kata-runtime.adoc:20]: @zanetworker This was from the initial commit. I haven't updated this section yet. So, if this content changed in the Google Doc it is not _yet_ reflected.

**Comment by zanetworker** [comment]
cc @c3d  https://github.com/openshift/openshift-docs/pull/32307#issuecomment-839736707

File: modules/sandboxed-containers-building-blocks.adoc
Comment (jensfr) [reply modules/sandboxed-containers-building-blocks.adoc:11]: @zanetworker AFAICS it will be registry.redhat.io/rh-osbs/openshift-tech-preview/sandboxed-containers-operator-bundle and registry.redhat.io/rh-osbs/openshift-tech-preview/sandboxed-containers-operator

File: modules/sandboxed-containers-rhcos-extensions.adoc
Comment (jensfr) [reply modules/sandboxed-containers-rhcos-extensions.adoc:9]: I would change it to "Kata and QEMU RPMs are installed as part of the sandoxed-containers RHCOS extension". It's only one extension and it contains the RPMs for kata, qemu-kiwi and its dependencies.

File: modules/sandboxed-containers-uninstalling-kata-runtime.adoc
Comment (jensfr) [reply modules/sandboxed-containers-uninstalling-kata-runtime.adoc:20]: @zanetworker @sagidlow ack from my side

File: sandboxed_containers/understanding-sandboxed-containers.adoc
Comment (jensfr) [inline sandboxed_containers/understanding-sandboxed-containers.adoc:21]: "to manage the lifecycle of sandboxed containers" sounds to me like the operator can start/stop/delete containers. But it doesn't control the sandboxed containers it only installs/uninstalls the runtime that is controlling the containers. I'd just remove the first sentence.

Comment (zanetworker) [reply sandboxed_containers/understanding-sandboxed-containers.adoc:21]: mmm, maybe just an overstatement for 4.8 but do you see it as a valid statement for the desired goal?

Comment (zanetworker) [reply sandboxed_containers/understanding-sandboxed-containers.adoc:21]: But I agree, we should reflect the install/un-install bit and revisit the "lifecycle" statement in 4.9

File: modules/sandboxed-containers-rhcos-extensions.adoc
Comment (sagidlow) [inline modules/sandboxed-containers-rhcos-extensions.adoc:9]: @jensfr I reworded a bit of what you wrote in your last comment. Does this look okay?
@zanetworker Please ack also.

File: modules/sandboxed-containers-building-blocks.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-building-blocks.adoc:11]: @jensfr, @zanetworker - Here are we trying to tell the user where they can download the Operator? Or are we trying to tell them the parts that make up the operator?

File: modules/sandboxed-containers-rhcos-extensions.adoc
Comment (jensfr) [reply modules/sandboxed-containers-rhcos-extensions.adoc:9]: @sagidlow looks good to me, thanks! just a nit, make kata -> Kata, and qemu-kiwi -> QEMU

Comment (sagidlow) [reply modules/sandboxed-containers-rhcos-extensions.adoc:9]: @jensfr , I just realized there is two sentences repeating the same thing >.< 

*{op-system} - RHCOS
*{op-system-first} is Red Hat Core OS 

Updating to read as follows:

The {sandboxed-containers-operator} is based on the {op-system-first} extensions concept. The sandboxed containers {op-system} extension contains RPMs for Kata, QEMU, and its dependencies. You can enable them by using the `MachineConfig` resources that the Machine Config Operator provides.

File: modules/sandboxed-containers-building-blocks.adoc
Comment (zanetworker) [reply modules/sandboxed-containers-building-blocks.adoc:11]: Its closer to the latter, basically how the operator deploys its content.

**Comment by cpmeadors** [comment]
I am providing QE approval for this section of the documentation.

File: modules/sandboxed-containers-building-blocks.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-building-blocks.adoc:11]: Rewrite suggestion: The {sandboxed-containers-operator} is constructed from the [sandboxed operator bundle](registry.redhat.io/rh-osbs/openshift-tech-preview/sandboxed-containers-operator-bundle) and the [container image](registry.redhat.io/rh-osbs/openshift-tech-preview/sandboxed-containers-operator). The images contains the controller part of the Operator that monitors and manages the `KataConfig` resource.

Does that sound write? @zanetworker @jensfr  -- Feel free to offer suggestions for rework.

File: modules/sandboxed-containers-common-terms.adoc
Comment (bpradipt) [inline modules/sandboxed-containers-common-terms.adoc:27]: nit: The RuntimeClass is named `kata`

File: modules/sandboxed-containers-building-blocks.adoc
Comment (jensfr) [reply modules/sandboxed-containers-building-blocks.adoc]: How about: `The {sandboxed-containers-operator} is packaged in the  [Operator bundle format](https://docs.openshift.com/container-platform/4.7/operators/operator_sdk/osdk-working-bundle-images.html). Therefore it comes in two container images. The Bundle image contains metadata and is required to make the operator OLM-ready. The second container image contains the actual controller that monitors and manages the KataConfig resource.`

File: modules/sandboxed-containers-common-terms.adoc
Comment (c3d) [reply modules/sandboxed-containers-common-terms.adoc:27]: @bpradipt: Good catch ;-)

File: modules/sandboxed-containers-building-blocks.adoc
Comment (c3d) [reply modules/sandboxed-containers-building-blocks.adoc]: I like that wording.

Comment (sagidlow) [reply modules/sandboxed-containers-building-blocks.adoc]: Made the changes based on @jensfr comment.

File: modules/sandboxed-containers-common-terms.adoc
Comment (bobfuru) [reply modules/sandboxed-containers-common-terms.adoc:17]: I haven't seen this succinct of a definition of our term `Operator` before in our docs, but I absolutely love it!

File: modules/sandboxed-containers-common-terms.adoc
Comment (bobfuru) [inline modules/sandboxed-containers-common-terms.adoc:23]: s/represents/represent

File: modules/sandboxed-containers-common-terms.adoc
Comment (bobfuru) [inline modules/sandboxed-containers-common-terms.adoc:27]: Is "the pod overhead" clear enough? I've not run across this usage before and can't tell if it means "overhead" as in "above" or as in "debt".

**Review by bobfuru — COMMENTED** [review]
A few comments and nits before merge, please. Thanks!

File: modules/common-attributes.adoc
Comment (sagidlow) [reply modules/common-attributes.adoc:30]: @bobfuru  the approved product name is "OpenShift sandboxed containers". We decided that we will use as above for our documentation to make sure that we are maintaining the approved product name.

File: modules/sandboxed-containers-common-terms.adoc
Comment (sagidlow) [reply modules/sandboxed-containers-common-terms.adoc:27]: I believe it is in terms of debt. @jensfr could you clarify?

Comment (c3d) [reply modules/sandboxed-containers-common-terms.adoc:27]: The term is a standard Kubernetes term with its own page. You could link to it: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/

**Comment by zanetworker** [comment]
@c3d Testing your OpenShift membership :) (Will remove later)

**Pull request merged**
